File: frontend-shared/api/locations.ts
/**
 * Saved locations API functions.
 * Manages the fishing spots a user has bookmarked for monitoring.
 */

import type { ApiClient } from './client';
import type {
  SavedLocation,
  SaveLocationRequest,
  UpdateLocationRadiusRequest,
} from '../types/location';

// Saves a new fishing spot for the user to monitor.
// After saving, new wildlife events within the location's radius will
// trigger push notifications.
export async function saveLocation(
  client: ApiClient,
  data: SaveLocationRequest,
): Promise<SavedLocation> {
  return client.post<SavedLocation>('/locations', data);
}

// Returns all the fishing spots the current user has saved.
// Used to populate the "My Spots" screen and the notification settings context.
export async function getUserLocations(
  client: ApiClient,
): Promise<SavedLocation[]> {
  return client.get<SavedLocation[]>('/locations');
}

// Removes a saved location. The user will no longer receive alerts
// based on this spot after deletion.
export async function deleteLocation(
  client: ApiClient,
  locationId: number,
): Promise<void> {
  return client.delete<void>(`/locations/${locationId}`);
}

// Updates the alert radius for an existing saved location.
// A larger radius casts a wider net for alerts; smaller means only very local events.
export async function updateLocationRadius(
  client: ApiClient,
  locationId: number,
  data: UpdateLocationRadiusRequest,
): Promise<SavedLocation> {
  return client.put<SavedLocation>(`/locations/${locationId}/radius`, data);
}

File: backend/handlers/locations.go
package handlers

import (
	"net/http"
)

// LocationHandler manages the fishing spots a user has saved for monitoring.
// Saved locations are what the notification job checks against when new
// wildlife events come in — if an event is within the radius of a saved spot,
// the user gets a push notification.
type LocationHandler struct {
	db interface{} // will be *pgxpool.Pool once implemented
}

// NewLocationHandler creates a new LocationHandler ready to use.
func NewLocationHandler(db interface{}) *LocationHandler {
	return &LocationHandler{db: db}
}

// SaveLocation stores a new fishing spot for the user.
// The user provides a name, GPS coordinates, and a radius in kilometers.
// After saving, the notification job will start checking this spot
// on every ingest cycle.
func (h *LocationHandler) SaveLocation(w http.ResponseWriter, r *http.Request) {
}

// GetUserLocations returns all the saved spots belonging to the logged-in user.
// This is what the "My Spots" screen reads to show the list of monitored locations.
func (h *LocationHandler) GetUserLocations(w http.ResponseWriter, r *http.Request) {
}

// DeleteLocation removes a saved spot. After deletion, the user will no longer
// receive alerts based on that location. Notifications already sent are not affected.
func (h *LocationHandler) DeleteLocation(w http.ResponseWriter, r *http.Request) {
}

// UpdateLocationRadius lets the user change how large an area around their saved
// spot they want to monitor. A smaller radius means only very local events trigger
// an alert; a larger one casts a wider net.
func (h *LocationHandler) UpdateLocationRadius(w http.ResponseWriter, r *http.Request) {
}
